hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieMemoryConfig.java

```diff
@@ -77,10 +80,19 @@ public class HoodieMemoryConfig extends HoodieConfig {
       .defaultValue(16 * 1024 * 1024)
       .withDocumentation("Property to control the max memory in bytes for dfs input stream buffer size");
 
-  public static final ConfigProperty<String> SPILLABLE_MAP_BASE_PATH = ConfigProperty
+  public static ConfigProperty<String> SPILLABLE_MAP_BASE_PATH = ConfigProperty
       .key("hoodie.memory.spillable.map.path")
       .defaultValue("/tmp/")
-      .withDocumentation("Default file path prefix for spillable map");
+      .withInferFunction(cfg -> {
+        // if user doesn't configure it, infer the path from env var
+        // if nothing to infer from, return Option.empty() then it will fallback to the default /tmp/
+        if (cfg.contains(HoodieTableConfig.SPILLABLE_MAP_BASE_PATH)) {
+          return Option.of(cfg.getString(HoodieTableConfig.SPILLABLE_MAP_BASE_PATH));
+        }
+        String[] localDirs = FileIOUtils.getConfiguredLocalDirs();
+        return (localDirs != null && localDirs.length > 0) ? Option.of(localDirs[0]) : Option.of("/tmp/");
```

Comment:
/nit return option.empty means using the default val

```suggestion
        return (localDirs != null && localDirs.length > 0) ? Option.of(localDirs[0]) : Option.empty();
```